install/self-node-remediation-deamonset.yaml

```diff
@@ -8,7 +8,7 @@ metadata:
 spec:
   selector:
     matchLabels:
-      control-plane: controller-manager
+      app: self-node-remediation-agent
```

Comment:
I'd suggest to use recommended label names, e.g.:

- `app.kubernetes.io/name: self-node-remediation`
- `app.kubernetes.io/component: agent`

See https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/

Maybe out of scope for this PR, but it might make sense to update all labels, see https://github.com/medik8s/node-healthcheck-operator/pull/146 as example.